hack/numaflow-controller-def-generator/numaflow-controller-def-generator.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+
+# This script uses Kustomize and yq to generate a templated Numaflow Controller definition manifest 
+# based on the latest version of the Numaflow numaspace-install manifest. The script can be called 
+# without arguments which will generate a file in the tests/manifests/default directory, or it 
+# can be called passing as argument a file path which will be used as the final output yaml.
+
+BASE_DIR=../..
+TEST_MANIFEST_DIR=tests/manifests/default
+SCRIPT_DIR=$(dirname "$0")
+
+# Change working directory
+cd $SCRIPT_DIR
+
+# Get the latest Numaflow Version from the remote repository
+echo "Getting the latest Numaflow version tag..."
+NUMAFLOW_VERSION=$(git ls-remote --tags https://github.com/numaproj/numaflow | grep -v '{}' | tail -1 | sed 's/.*\///' | sed 's/^v//')
```

Comment:
Ideally we should be able to run this script for any Git tag and not just the latest, right? (but I'm fine if done in a follow up)

Reply:
I'll add an argument to use a specific tag if present or to use latest if not present.